src/test/java/com/wavefront/WavefrontNozzleApplicationTests.java

```diff
@@ -1,17 +1,79 @@
 package com.wavefront;
 
+import com.wavefront.props.FirehoseProperties;
+import com.wavefront.proxy.ProxyForwarder;
+import com.wavefront.proxy.ProxyForwarderImpl;
+import com.wavefront.service.FirehoseToWavefrontProxyConnector;
+import org.cloudfoundry.doppler.DopplerClient;
+import org.cloudfoundry.doppler.Envelope;
+import org.cloudfoundry.doppler.EventType;
+import org.cloudfoundry.dropsonde.events.CounterEvent;
+import org.easymock.EasyMock;
 import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.test.context.junit4.SpringRunner;
+import reactor.core.publisher.Flux;
 
-@RunWith(SpringRunner.class)
-@SpringBootTest
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * Wavefront Firehose Nozzle Unit Tests
+ *
+ * @author Sushant Dewan (sushant@wavefront.com).
+ */
 public class WavefrontNozzleApplicationTests {
 
-	@Test
-	public void contextLoads() {
-		// TODO ...
-	}
+  @Test
+  public void testFirehoseConnectorProxyForwarder() throws IOException, InterruptedException {
+    FirehoseProperties firehoseProperties = new FirehoseProperties();
+    firehoseProperties.setSubscriptionId("subscriptionId");
+    firehoseProperties.setEventTypes(Arrays.asList(EventType.COUNTER_EVENT,
+            EventType.VALUE_METRIC, EventType.CONTAINER_METRIC));
+
+    testInternal(firehoseProperties, "CounterEvent",
+            org.cloudfoundry.dropsonde.events.Envelope.EventType.CounterEvent);
```

Comment:
indenting is still odd, check to see your IDE setting for this project is the same as the system